### PR TITLE
Conditional stylecop activation

### PR DESCRIPTION
**Pull request by Frassle**
### Purpose of this PR

Tidy up of #675.

It's not needed to run StyleCop rules for every build. They take a long time (~13 minutes) but we want to ensure they are run before merge (so appveyvor and travis still run them).

### CODE REVIEW

**Comment by Frassle** [comment]
Well looks like I turned on checking for the GLControl project. Surprised that wasn't on already, will fix up in this PR.

**Comment by VPeruS** [comment]
Very suspicious build time 1 min instead of previous 15

**Comment by Frassle** [comment]
Was a typo in the travis file

**Comment by Frassle** [comment]
Ok travis seems to be working. appveyor isn't picking up the env var.

**Comment by VPeruS** [comment]
https://ci.appveyor.com/project/varon/opentk/build/0.0.1.458, this is from another pull.
Seems appveyor provide more powerfull build system

**Comment by Frassle** [comment]
Ah ok appveyor is just fast :smile:

File: src/Generator.Converter/Generator.Convert.csproj [hunk 1]
```diff
@@ -157,14 +155,7 @@
       </ItemGroup>
     </When>
   </Choose>
-  <ItemGroup>
-    <Analyzer Include="..\..\packages\StyleCop.Analyzers\analyzers\dotnet\cs\StyleCop.Analyzers.CodeFixes.dll">
```
Comment (Nihlus) [inline src/Generator.Converter/Generator.Convert.csproj]: All of these changes are going to be reverted the moment anyone runs paket.

**Comment by VPeruS** [comment]
With or without env variable appveyor use 4-5 mins to build, look at #675's first successful CI commit.

**Comment by Frassle** [comment]
> instead of disabling stylecop altogether, the autogenerated files are excluded?

Probably significant but that's probably better done by the project split up so we don't have to build all the autogenned files every time.

> This looks good overall, but I worry that we'll be seeing a lot more broken PRs

Maybe leave it as default enabled, but use the envvar to turn it off. Then can be used by experienced users to improve iteration time without new users not running it?

**Comment by Nihlus** [comment]
Ah, no, I'm referring to marking the generated files with an autogenerated tag, which would make stylecop ignore them. 

I agree that they should be default on.

**Comment by Frassle** [comment]
Ok I've made it so by default stylecop is ON. I'll amend the wiki to point out it can be turned of for quicker builds.

**Comment by Frassle** [comment]
Just need to sort out so paket doesn't overwrite everything

**Comment by Nihlus** [comment]
Looks good to merge now.

**Comment by Frassle** [comment]
Ah I hadn't rerun paket yet... Will add another PR, will see if I can add a check to travis/appveyor that it doesn't happen again as well.

**Pull request merged**